content/articles/graphs-in-data-structure -using _c++/index.md

```diff
@@ -0,0 +1,153 @@
+
+We can store the cost of the edge at A<sub>i,j</sub> in the adjacency matrix of a weighted graph rather than storing 0 or 1.
+
+If A<sub>i,j</sub> is 1 in undirected graph then A<sub>i,j</sub> will also be 1.
+
+If A<sub>i,j</sub> is 1 in  directed graph, then may or may not be 1.
+
+The adjacency matrix offers constant-time access (O(1)) to detect if two nodes have an edge. Therefore, the adjacency matrix has a space complexity of O(V)<sup>2</sup>.
+
+* **Adjacency List**
+ It is the most common way of representing a graph when we implement it in our code. An adjacency list is a collection of distinct lists in an array A. For example, every node of a graph A<sub>i</sub> is a list that holds all the vertices' values next to vertex i.
+
+If we have a weighted graph, we store the cost with the vertex using pairs. You will get a better understanding when we implement it in our code. If vertex j is in list A<sub>i</sub>, vertex i will be in list A<sub>j</sub> in an undirected graph. 
+
+Due to the fact that information in the adjacency list is only stored for edges that exist in the graph, its space complexity is O(V + E).
+Representation of an undirected graph,
+
+* G is an undirected graph with 5 vertices and 7 edges.
+  
+  ![undirected-graph](/engineering-education/graphs-in-data-structure -using _c+/graph_6.jpg)
+
+* Adjacency-list representation of G
+  
+  ![graphList](/engineering-education/graphs-in-data-structure -using _c+/graph_7.jpg)
+
+* The adjacency-matrix representation of G
+  
+  ![graphMartrix](/engineering-education/graphs-in-data-structure -using _c+/graph_8.jpg)
+
+**Code for Adjacency list representation of graph** : 
+
+``` C++
+#include <iostream>
+#include<list>
+using namespace std;
+
+class graph{
+public:
+    list<int> *adjlist;
+    int n;
+    graph(int v){
+        adjlist=new list<int> [v];
+        n=v;
+    }
+
+    void addedge(int u,int v,bool bi){
+        adjlist[u].push_back(v);
+        if(bi){
+            adjlist[v].push_back(u);
+        }
+    }
+
+    void print(){
+        for(int i=0;i<n;i++){
+            cout<<i<<"-->";
+            for(auto it:adjlist[i]){
+                cout<<it<<" ";
+            }
+            cout<<endl;
+        }
+        cout<<endl;
+    }
+
+
+};
+
+int main() {
+    graph g(5);
+    g.addedge(1,2,true);
+    g.addedge(4,2,true);
+    g.addedge(1,3,true);
+    g.addedge(4,3,true);
+    g.addedge(1,4,true);
+
+    g.print();
+}
+
+```
+
+You can find the complete execution of the above code [here](https://replit.com/@AnubhavBansal1/Gaphs#main.cpp).
+
+#### Output
+
+![output](./output.jpg)
+
+* In the output you can see the implementation of graphs in adjacency list view.
+
+#### Conclusion
+
+By learning graphs from basics you will get to think more analytically while solving data structures questions. It would be great if you practice some of the graphs questions after understanding the foundation of graphs 
```

Comment:
Change to " By learning graphs from basics, you will get to think more analytically while solving data structures questions. It would be great if you practice some of the graphs questions after understanding the foundation of graphs "